SIPS/sip-30.md

```diff
@@ -0,0 +1,238 @@
+---
+sip: 30
+title: Entropy Source Identifiers
+status: Draft
+author: Shane T Odlum (@shane-t)
+created: 2024-12-17
+---
+
+## Abstract
+
+This SIP proposes additions to entropy retrieval APIs that allows snaps to request entropy from a specific source.
+
+## Motivation
+
+Interoperability snaps and account management snaps use the methods `snap_getEntropy`, `snap_getBip44Entropy`, `snap_getBip32Entropy`, and `snap_getBip32PublicKey` to generate addresses and other key material.
+
+These methods assume the client contains a single entropy source (the user's primary keyring mnemonic). The proposed API changes will allow snaps to request entropy from a specific source such as a secondary mnemonic. A new method `snap_listAvailableEntropySources` will be added to allow snaps to request a list of available entropy sources.
+
+## Specification
+
+> Formal specifications are written in TypeScript.
+
+### Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",
+"NOT RECOMMENDED", "MAY", and "OPTIONAL" written in uppercase in this document are to be interpreted as described in
+[RFC 2119](https://www.ietf.org/rfc/rfc2119.txt)
+
+### Snap Manifest
+
+A new set permission is added to the snap manifest:
+
+```json
+{
+  "initialPermissions": {
+    "snap_listAvailableEntropySources": {}
+  }
+}
+```
+
+
+### Common Types
+```typescript
+type Slip10Node = {
+  depth: number;
+  parentFingerprint: number;
+  index: number;
+  privateKey: string;
+  publicKey: string;
+  chainCode: string;
+  curve: "ed25519" | "secp256k1";
+};
+
+export type BIP44Node = {
+  coin_type: number;
+  depth: number;
+  privateKey: string;
+  publicKey: string;
+  chainCode: string;
+  path: string[];
+};
+
+interface EntropySource {
```

Comment:
Should we flag to the Snap which one is the primary one?

Reply:
I have added a simple boolean flag here